packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-slug.js

```diff
@@ -87,6 +87,12 @@ export default function PageSlug( { postType, postId } ) {
 				popoverProps={ popoverProps }
 				focusOnMount
 				ref={ setPopoverAnchor }
+				onClose={ () => {
+					if ( forceEmptyField ) {
+						onSlugChange( cleanForSlug( savedSlug ) );
+						setForceEmptyField( false );
+					}
+				} }
```

Comment:
You can close the modals via the Esc key; the action doesn't trigger the blur event, so the value isn't reverted, and the field stays empty.

## Screencast

https://github.com/WordPress/gutenberg/assets/240569/20aefbe1-3ea4-48f4-8b2b-16a2d85105c0